users/danielo515/alt_tab.c

```diff
@@ -0,0 +1,38 @@
+#include "danielo515.h"
+#include "alt_tab.h"
+
+bool                       altPressed = false;
+__attribute__((weak)) void alt_tab_activated(void){};
+__attribute__((weak)) void alt_tab_deactivated(void){};
+extern bool                onMac;
+
+// =============== ALT_TAB single key handling
+bool process_alt_tab(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+        case ALT_TAB:
+            if (!record->event.pressed) {
+                return false;
+            }
+            if (altPressed) {
+                tap_code(KC_TAB);
+            } else {
+                altPressed = true;
+                onMac ? register_code(KC_LGUI) : register_code(KC_LALT);
```

Comment:
If you use the alt-gui swap stuff, you could just use:
```suggestion 
                register_code(keycode_config(KC_LALT));
```
`keycode_config` actually handles this check, based on the global config.

https://github.com/qmk/qmk_firmware/blob/f76f9c7d2abd1901be45431c5b2ea435c8c472b2/quantum/keycode_config.c#L45-L63

Reply:
If I understand correctly, the keymap_config is changed by bootmagic. In my case, I'm changing this flag "on the fly". This means, that I just hit a key or trigger certain leader combo and the flag flips to the opposite value. I don't think this is possible with bootmagic right?

Reply:
There are keycodes available that effectively do the same thing as your `MAC_TGL`: https://docs.qmk.fm/#/keycodes?id=bootmagic

You don't need Bootmagic enabled at all to use them; in fact it is technically a separate feature.

Reply:
Ok, makes sense. I took a briefly overview after @drashna 's comment and I deduced that same thing, but thanks for confirmation. In any case, seems to be a bit overkill for my use case. All I do is a single flag toggle and then I use it for all the things that needs to be adjusted: macros will send the correct modifier which can vary from control, to alt as an alternative to CMD. The proposed feature seems to be a lot more fine grained. I don't want to think about switching CMD by CTL or if I prefer to use ALT instead of CMD, what I want is all my keyboard features to be aware of the different operating system and adjust as needed. Such adjustment may vary from case to case.

In any case, thank you for discovering to me another of the endless features QMK has.

Reply:
Well, that's the thing, the "magic" feature is basically just a simple toggle, with the handling for it too.  

You don't have to use it, but I figured it was worth mentioning, since you're essentially reimplementing a feature that already exists.

Reply:
I don't see it as the same feature. Sure they overlap, but the use case is not the same. I just need to handle one toggle, and all my macros and keycodes will adjust automatically. With the suggested feature I have to think which key I want to toggle by which one.